bin/brew

```diff
@@ -9,20 +9,43 @@ BREW_FILE_DIRECTORY="$(quiet_cd "${0%/*}/" && pwd -P)"
 HOMEBREW_BREW_FILE="${BREW_FILE_DIRECTORY%/}/${0##*/}"
 
 HOMEBREW_PREFIX="${HOMEBREW_BREW_FILE%/*/*}"
-[[ -n "$HOMEBREW_PREFIX" && "$HOMEBREW_PREFIX" != "$HOMEBREW_BREW_FILE" ]] \
-  || HOMEBREW_PREFIX="/"
+if [[ -z "$HOMEBREW_PREFIX" || "$HOMEBREW_PREFIX" == "$HOMEBREW_BREW_FILE" ]]
+then
+  HOMEBREW_PREFIX="/"
+fi
 
 HOMEBREW_REPOSITORY="$HOMEBREW_PREFIX"
 
 if [[ -L "$HOMEBREW_BREW_FILE" ]]
 then
+  HOMEBREW_BREW_FILE_LINK="1"
+fi
+
+if [[ -L "/usr/local/bin/brew" ]]
+then
+  HOMEBREW_USR_LOCAL_HOMEBREW_TEST="1"
+fi
+
+if [[ -n "$HOMEBREW_BREW_FILE_LINK" ||
+      -n "$HOMEBREW_USR_LOCAL_HOMEBREW_TEST" ]]
+then
   BREW_SYMLINK="$(readlink "$HOMEBREW_BREW_FILE")"
   BREW_SYMLINK_DIRECTORY="$(dirname "$BREW_SYMLINK")"
   BREW_FILE_DIRECTORY="$(quiet_cd "$BREW_FILE_DIRECTORY" &&
                          quiet_cd "$BREW_SYMLINK_DIRECTORY" && pwd -P)"
+fi
+
+if [[ -n "$HOMEBREW_BREW_FILE_LINK" ]]
+then
   HOMEBREW_REPOSITORY="${BREW_FILE_DIRECTORY%/*}"
 fi
 
+if [[ -n "$HOMEBREW_USR_LOCAL_HOMEBREW_TEST" &&
+      "$BREW_FILE_DIRECTORY" == "$HOMEBREW_REPOSITORY/bin" ]]
```

Comment:
Also, the second part of the test (the `"$BREW_FILE_DIRECTORY" = "$HOMEBREW_REPOSITORY/bin"`) is always true unless I'm missing something.